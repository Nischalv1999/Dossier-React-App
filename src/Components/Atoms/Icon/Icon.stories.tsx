import React from "react";
import { ComponentStory, ComponentMeta, Story } from "@storybook/react";
import Icon, { iconProps } from "./Icon";
import { ComponentProps } from "react";

import themes from 'assets/typography.svg'


export default {
  title: "Atoms/Icons",
  component: Icon,
} 


const Template: Story<ComponentProps<typeof Icon>> = (args:iconProps) => (
    <Icon {...args} />
  )
  
  export const themeIcon = Template.bind({})
  
  themeIcon.args = {
   imageSrc:themes
  }