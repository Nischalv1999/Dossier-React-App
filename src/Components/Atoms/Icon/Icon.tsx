import React from "react";
import { ReactSVG } from 'react-svg';


export type iconProps=
{
imageSrc:string;
classname?:any;

}

const Icon = (props:iconProps)=> {

    const { imageSrc, classname }= props

    return(
        <ReactSVG {...props} src={imageSrc} className={classname}/>
    )
}

export default Icon;