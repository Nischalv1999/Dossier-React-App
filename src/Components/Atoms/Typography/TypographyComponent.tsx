import { Typography } from "@mui/material";
import * as React from "react";

export type TypographyProps = {
  variant?:
    | "h1"
    | "h2"
    | "h3"
    | "h4"
    | "h5"
    | "h6"
    | "subtitle1"
    | "subtitle2"
    | "body1"
    | "body2"
    | "caption"
    | "button"
    | "overline"
    | "inherit";
  text?: string;
  color?: string;
  classname?:string
};

const TypographyComponent: React.FC<TypographyProps> = ({
  variant,
  text,
  color,
  classname
}) => {
  return (
    <Typography variant={variant} color={color} className={classname}>
      {text}
    </Typography>
  );
};

export default TypographyComponent;
