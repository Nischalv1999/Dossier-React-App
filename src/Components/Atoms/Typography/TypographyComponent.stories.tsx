import { ComponentProps } from "react";

import { Story } from "@storybook/react";
import TypographyComponent, { TypographyProps } from "./TypographyComponent";
import { ThemeProvider } from "@mui/material";
import theme from "../../../theme/theme";

export default {
  title: "Atoms/Typography",
  component: TypographyComponent,
};

const Template: Story<ComponentProps<typeof TypographyComponent>> = (
  args: TypographyProps
) => (
  <ThemeProvider theme={theme}>
    <TypographyComponent {...args} />
  </ThemeProvider>
);

export const Text = Template.bind({});

Text.args = {
  variant: "h3",
  text: "This is Text",
  color: "#4C2CD9",
};
