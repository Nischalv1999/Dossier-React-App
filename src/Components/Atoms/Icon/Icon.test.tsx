import { describe, it } from "@jest/globals";

import React from "react";
import { render } from "@testing-library/react";
import "@testing-library/jest-dom";
import Icon from "./Icon";
import Typography from "assets/typography.svg";


describe("Icons test", () => {
  it("Matches the snapshot", () => {
    const { container } = render(<Icon imageSrc={Typography} />);
    expect(container).toMatchSnapshot();
  });
});
