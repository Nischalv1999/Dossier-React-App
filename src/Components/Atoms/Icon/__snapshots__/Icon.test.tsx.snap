// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Icons test Matches the snapshot 1`] = `
<div>
  <div
    imagesrc="typography.svg"
  >
    <div>
      <div
        data-src="typography.svg"
        src=""
      />
    </div>
  </div>
</div>
`;
